### PR TITLE
Add support for "User Default" settings

### LINKED ISSUE #2325: Default Profile for Common Profile Settings
# Default Profile for Common Profile Settings

Allow settings which the user would like applied to all profiles to be specified in a profile with a well-known name, such as `"name" : "_default"`. All settings in this profile will be applied before they are overridden from individual profiles. This would be helpful to allow all profiles to have acrylic applied with a certain opacity where those settings are not specified.

    [snip]
    {
        "name" : "_default",
        "useAcrylic" : true,
        "acrylicOpacity" : 0.5,
        "closeOnExit" : true,
        "colorScheme" : "Campbell",
        "cursorColor" : "#FFFFFF",
        "cursorShape" : "bar",
        "fontFace" : "Consolas",
        "fontSize" : 10,
        "historySize" : 9001,
        "padding" : "0, 0, 0, 0",
        "snapOnInput" : true,
        "startingDirectory" : "%USERPROFILE%"
    },
    {
        "name" : "PowerShell Core",
        "commandline" : "C:\\Program Files\\PowerShell\\6\\pwsh.exe",
        "guid" : "{574e775e-4f2a-5b96-ac1e-a2962a402336}",
        "icon" : "ms-appx:///ProfileIcons/{574e775e-4f2a-5b96-ac1e-a2962a402336}.png",
    },
    {
        "name" : "cmd",
        "commandline" : "cmd.exe",
        "guid" : "{0caa0dad-35be-5f56-a8ff-afceeeaa6101}",
        "icon" : "ms-appx:///ProfileIcons/{0caa0dad-35be-5f56-a8ff-afceeeaa6101}.png",
        "useAcrylic" : false
    },
    [snip]

### PR DESCRIPTION
**Pull request by zadjii-msft**
## Summary of the Pull Request

These are settings that apply to _every_ profile, before user customizations. This can be placed in the globals under the "defaultSettings" key. It's a blob of profile settings that will be parsed just like a profile, and applied to profiles after default&dynamic profiles, but before user profiles.

"defaultSettings" is kinda a placeholder key ATM - @cinnamon-msft do we have concerns with this name? Does it make enough sense?

## References
#2515, #2603 

## PR Checklist
* [x] Closes #2325
* [x] I work here
* [x] Tests added/passed
* [x] schema, docs updated

## Detailed Description of the Pull Request / Additional comments
Discussion in #2325 itself serves as the "spec" for this task. I thought we'd need more discussion on the topic, but it ended up being pretty straightforward.

## Validation Steps Performed
_ran the tests_

### CODE REVIEW

File: doc/cascadia/profiles.schema.json [hunk 1]
```diff
@@ -155,263 +159,267 @@
...
-                },
-                "white": {
-                  "$ref": "#/definitions/Color",
-                  "description": "Sets the color used as ANSI white."
-                },
-                "yellow": {
-                  "$ref": "#/definitions/Color",
-                  "description": "Sets the color used as ANSI yellow."
-                }
+    "Profile": {
```
Comment (DHowett-MSFT) [inline doc/cascadia/profiles.schema.json]: this schema probably says name or guid is required for defaultsettings, eh. i wonder if there's some all_of or any_of magic we can use to make sure that some profiles have guids+settings and some just have settings? don't necessarily block on this ..

Comment (zadjii-msft) [reply doc/cascadia/profiles.schema.json]: I already fixed that :)
Profiles in the "defaultSettings" don't have anything required, but ones in ProfileList do

Comment (DHowett-MSFT) [reply doc/cascadia/profiles.schema.json]: Oo, i see

File: doc/cascadia/SettingsSchema.md [hunk 2]
```diff
@@ -8,6 +8,7 @@ Properties listed below affect the entire window, regardless of the profile sett
 | `alwaysShowTabs` | _Required_ | Boolean | `true` | When set to `true`, tabs are always displayed. When set to `false` and `showTabsInTitlebar` is set to `false`, tabs only appear after typing <kbd>Ctrl</kbd> + <kbd>T</kbd>. |
 | `copyOnSelect` | Optional | Boolean | `false` | When set to `true`, a selection is immediately copied to your clipboard upon creation. When set to `false`, the selection persists and awaits further action. |
 | `defaultProfile` | _Required_ | String | PowerShell guid | Sets the default profile. Opens by typing <kbd>Ctrl</kbd> + <kbd>T</kbd> or by clicking the '+' icon. The guid of the desired default profile is used as the value. |
+| `defaultSettings` | Optional | Object (Profile) | | Default settings to use with each profile. Settings in each profile in profiles will override these values. |
```
[the remark this answers is not in the document]
Comment (carlos-zamora) [reply doc/cascadia/SettingsSchema.md]: tagging @cinnamon-msft

**Comment by zadjii-msft** [comment]
There's been quite the discussion on the team regarding this issue. There doesn't seem to be any sort of consensus on naming or how exactly this should work, so I'm closing this in favor of having a proper full spec review.

Functionally, the code change is simple, as seen here. Naming, however, is a Hard problem.

**Pull request closed without merging**